packages/authentication/src/__tests__/fixtures/services/jwt-service.ts

```diff
@@ -0,0 +1,86 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/authentication
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {inject} from '@loopback/context';
+import {HttpErrors} from '@loopback/rest';
+import {promisify} from 'util';
+import {TokenService} from '../../../services/token.service';
+import {UserProfile} from '../../../types';
+import {JWTAuthenticationStrategyBindings} from '../keys';
+const jwt = require('jsonwebtoken');
+const signAsync = promisify(jwt.sign);
+const verifyAsync = promisify(jwt.verify);
+
+export class JWTService implements TokenService {
+  constructor(
+    @inject(JWTAuthenticationStrategyBindings.TOKEN_SECRET)
+    private jwt_secret: string,
+    @inject(JWTAuthenticationStrategyBindings.TOKEN_EXPIRES_IN)
+    private jwt_expiresIn: string,
+  ) {}
+
+  async verifyToken(token: string): Promise<UserProfile> {
+    if (!token) {
+      throw new HttpErrors['Unauthorized'](
+        `Error verifying token : 'token' is null`,
+      );
+    } //if
+
+    let userProfile: UserProfile;
+
+    try {
+      // decode user profile from token
+      userProfile = await verifyAsync(token, this.jwt_secret);
+    } catch (error) {
+      throw new HttpErrors['Unauthorized'](`Error verifying token : ${error}`);
+    }
+
+    return userProfile;
+  }
+
+  async generateToken(userProfile: UserProfile): Promise<string> {
+    if (!userProfile) {
+      throw new HttpErrors['Unauthorized'](
+        'Error generating token : userProfile is null',
+      );
+    } //if
+
+    if (!userProfile.id) {
+      throw new HttpErrors['Unauthorized'](
+        `Error generating token : userProfile 'id'  is null`,
+      );
+    } //if
+
+    if (!userProfile.email) {
```

Comment:
I remember the `email` and `name` are optional, see [interface UserProfile](https://github.com/strongloop/loopback-next/blob/master/packages/authentication/src/types.ts#L20)

Reply:
@jannyHou:

 I realize that `email` and `name` can be optional in the `UserProfile` interface we have defined in the authentication package, but this is mainly so we don't force the community to use the two fields we are providing as `suggestions`.

 In my acceptance tests  in the `src/__tests__`folder, I create mock implementation classes for two strategies, for a token service, and for a user service, and I personally decided that each field in `UserProfile` (optional or not) is required for creating a JWT Token. The purpose of the mocha tests are to test the authentication strategy interface, and registering authentication strategies via the extensionPoint/extensions pattern.

Reply:
> and I personally decided that each field in UserProfile (optional or not) is required for creating a JWT Token. 

Fair enough 👍